pkg/auth/service.go

```diff
@@ -959,3 +964,626 @@ func (s *DBAuthService) Authorize(ctx context.Context, req *AuthorizationRequest
+			return policies, err
+		})
+		if err != nil {
+			return nil, nil, err
+		}
+		return policies, &model.Paginator{Amount: len(policies)}, nil
+	}
+	return a.listUserPolicies(ctx, username, params, true)
+}
+
+func (a *APIAuthService) AttachPolicyToGroup(ctx context.Context, policyDisplayName, groupDisplayName string) error {
+	resp, err := a.apiClient.AttachPolicyToGroupWithResponse(ctx, groupDisplayName, policyDisplayName)
+	if err != nil {
+		return err
+	}
+	return expectedStatusCode(resp, 201)
+}
+
+func (a *APIAuthService) DetachPolicyFromGroup(ctx context.Context, policyDisplayName, groupDisplayName string) error {
+	resp, err := a.apiClient.DetachPolicyFromUserWithResponse(ctx, groupDisplayName, policyDisplayName)
+	if err != nil {
+		return err
+	}
+	return expectedStatusCode(resp, 204)
+}
+
+func (a *APIAuthService) ListGroupPolicies(ctx context.Context, groupDisplayName string, params *model.PaginationParams) ([]*model.Policy, *model.Paginator, error) {
+	resp, err := a.apiClient.ListGroupPoliciesWithResponse(ctx, groupDisplayName, &ListGroupPoliciesParams{
+		Prefix: paginationPrefix(params.Prefix),
+		After:  paginationAfter(params.After),
+		Amount: paginationAmount(params.Amount),
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	if err := expectedStatusCode(resp, 200); err != nil {
+		return nil, nil, err
+	}
+	policies := make([]*model.Policy, len(resp.JSON200.Results))
+
+	for i, r := range resp.JSON200.Results {
+		policies[i] = serializePolicyToModalPolicy(r)
+	}
+	return policies, toPagination(resp.JSON200.Pagination), nil
+}
+
+func (a *APIAuthService) Authorize(ctx context.Context, req *AuthorizationRequest) (*AuthorizationResponse, error) {
+	policies, _, err := a.ListEffectivePolicies(ctx, req.Username, &model.PaginationParams{
+		After:  "", // all
+		Amount: -1, // all
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	allowed := checkPermissions(req.RequiredPermissions, req.Username, policies)
+
+	if allowed != CheckAllow {
+		return &AuthorizationResponse{
+			Allowed: false,
+			Error:   ErrInsufficientPermissions,
+		}, nil
+	}
+
+	// we're allowed!
+	return &AuthorizationResponse{Allowed: true}, nil
+}
+
+func NewAPIAuthService(apiEndpoint, token string, secretStore crypt.SecretStore, cacheConf params.ServiceCache) (*APIAuthService, error) {
+	// TODO(Guys): currently token is used, maybe we should consider using APIKey
+	basicAuthProvider, err := securityprovider.NewSecurityProviderBearerToken(token)
+	if err != nil {
+		return nil, err
+	}
+	httpClient := &http.Client{
+		// avoid redirect automatically
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+	client, err := NewClientWithResponses(
+		apiEndpoint,
+		WithRequestEditorFn(basicAuthProvider.Intercept),
+		WithHTTPClient(httpClient),
+	)
+	if err != nil {
+		return nil, err
+	}
+	logging.Default().Info("initialized Auth service")
```

Comment:
consider accepting logger - or initialize one here with field "service" set to remote authorization service